src/main/java/org/fcrepo/migration/validator/impl/ValidatingObjectHandler.java

```diff
@@ -0,0 +1,233 @@
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * A streaming object handler implementation that performs object scoped validations on behalf
+ * of the Fedora3ObjectValidator.
+ *
+ * @author dbernstein
+ */
+public class ValidatingObjectHandler implements FedoraObjectHandler {
+    private static final Logger LOGGER = getLogger(Fedora3ObjectValidator.class);
+
+    private ObjectInfo objectInfo;
+    private OcflObjectSession ocflSession;
+    private final List<ValidationResult> validationResults = new ArrayList<>();
+    private int indexCounter;
+    private Set<String> headDatastreamIds = new HashSet<>();
+
+    private static final Map<String, PropertyResolver> OCFL_PROPERTY_RESOLVERS = new HashMap<>();
+
+    static {
+        OCFL_PROPERTY_RESOLVERS.put("info:fedora/fedora-system:def/model#createdDate",
+                new PropertyResolver<String>() {
+                    @Override
+                    public String resolve(final ResourceHeaders headers) {
+                        return headers.getCreatedDate().toString();
+                    }
+                });
+    }
+
+    private interface PropertyResolver<T> {
+        T resolve(ResourceHeaders headers);
+    }
+
+
+    @Override
+    public void processObject(final ObjectReference objectReference) {
+        this.objectInfo = objectReference.getObjectInfo();
+        LOGGER.debug("beginning processing on object: pid={}", this.objectInfo);
+        if (initialObjectValidation(objectReference.getObjectProperties())) {
+            objectReference.listDatastreamIds().forEach(dsId -> validateDatastream(dsId, objectReference));
+            completeObjectValidation(objectReference.getObjectInfo());
+        }
+    }
+
+    /**
+     * Constructor
+     *
+     * @param session
+     */
+    public ValidatingObjectHandler(final OcflObjectSession session) {
+        this.ocflSession = session;
+    }
+
+    /**
+     * Result the validation results after processObject has been called.
+     *
+     * @return
+     */
+    public List<ValidationResult> getValidationResults() {
+        return validationResults;
+    }
+
+    /**
+     * @param objectProperties
+     * @return true if initial validation successful and should proceed.
+     */
+    private boolean initialObjectValidation(final ObjectProperties objectProperties) {
+        final var pid = objectInfo.getPid();
+        final var ocflId = ocflSession.ocflObjectId();
+
+        final ResourceHeaders headers;
+
+        try {
+            headers = ocflSession.readHeaders(ocflId);
+        } catch (NotFoundException ex) {
+            validationResults.add(new ValidationResult(indexCounter++, FAIL, OBJECT, SOURCE_OBJECT_EXISTS_IN_TARGET,
+                    pid, ocflId, "Source object not present in target repository."));
+            return false;
+        }
+
+        //check that last updated date:
+        objectProperties.listProperties().stream().forEach(op -> {
```

Comment:
I believe you can drop the `.stream()` call here.

Reply:
+1
```suggestion
        objectProperties.listProperties().stream().forEach(op -> {
```